src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
On 10/04/15 06:07, Matt Broadstone wrote:
> In src/qamqpclient.cpp
> <https://github.com/mbroadst/qamqp/pull/26#discussion_r28097583>:
> 
>> @@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
>>  QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
>>  {
>>      Q_D(QAmqpClient);
>> +    if (!name.isEmpty() && d->exchanges.contains(name)
>> +                    && (!d->exchanges[name].isNull()))
> 
> yeah I was pondering this earlier. I think it might be a decent idea to
> hook up that signal <http://doc.qt.io/qt-5/qobject.html#destroyed> to go
> through and clean the list. This is more of a consideration for
> longer-running servers, and I think perhaps maybe one worth pursuing.
> Thoughts?

Certainly.  The application I'm putting QAMQP here is for comms drivers,
so think, months of continuous uptime, an area where memory leaks will
definitely be felt.

I'll probably leave that check for isNull though for safety: if the
clean-up does its job, it's harmless, if it doesn't, it'll be handled.
-- 
Stuart Longland (aka Redhatter, VK4MSL)

I haven't lost my mind...
  ...it's backed up on a tape somewhere.